packages/astro/src/vite-plugin-astro-server/plugin.ts

```diff
@@ -50,26 +50,24 @@ export default function createVitePluginAstroServer({
 			const controller = createController({ loader });
 			const localStorage = new AsyncLocalStorage();
 
-			/** rebuild the route cache + manifest, as needed. */
-			async function rebuildManifest(needsManifestRebuild: boolean) {
+			/** rebuild the route cache + manifest */
+			async function rebuildManifest() {
 				pipeline.clearRouteCache();
-				if (needsManifestRebuild) {
-					routeManifest = injectDefaultDevRoutes(
-						settings,
-						devSSRManifest,
-						await createRouteManifest({ settings, fsMod }, logger), // TODO: Handle partial updates to the manifest
-					);
-					warnMissingAdapter(logger, settings);
-					pipeline.manifest.checkOrigin =
-						settings.config.security.checkOrigin && settings.buildOutput === 'server';
-					pipeline.setManifestData(routeManifest);
-				}
+				routeManifest = injectDefaultDevRoutes(
+					settings,
+					devSSRManifest,
+					await createRouteManifest({ settings, fsMod }, logger), // TODO: Handle partial updates to the manifest
+				);
+				warnMissingAdapter(logger, settings);
+				pipeline.manifest.checkOrigin =
+					settings.config.security.checkOrigin && settings.buildOutput === 'server';
+				pipeline.setManifestData(routeManifest);
 			}
 
-			// Rebuild route manifest on file change, if needed.
-			viteServer.watcher.on('add', rebuildManifest.bind(null, true));
-			viteServer.watcher.on('unlink', rebuildManifest.bind(null, true));
-			viteServer.watcher.on('change', rebuildManifest.bind(null, false));
+			// Rebuild route manifest on file change
+			viteServer.watcher.on('add', rebuildManifest.bind(null));
+			viteServer.watcher.on('unlink', rebuildManifest.bind(null));
+			viteServer.watcher.on('change', rebuildManifest.bind(null));
```

Comment:
I get why the optimization was removed, but isn't it too drastic?

The entire route manifest will be recomputed on every change anywhere. Even tho it will most likely not change anything in the routes.
- `add` changes routes, that's fine
- `unlink` changes routes, that's fine
- `change` only changes routes if the change adds, removes, or alters the `export const prerender` line.

Editing content will repeatedly recompute the routes, which includes reading all the files in the routes for the pretender parameter. That will make the dev server quite inefficient.

I think we should at least check to make sure that changes to files outside of the pages directory won't recompute the routes so that editing content doesn't consume that much extra resource on every save. I think the event value provided by the Chokidar API includes which file was changed

Reply:
Maybe now is the time to address the todo comment then, I'll look into that

Reply:
Maybe we should revert the change _for now_, and evaluate to send another PR just for the optimisation.